pkg/scheduler/reconciliation/filter.go

```diff
@@ -8,12 +8,106 @@ import (
 	"github.com/kyma-incubator/reconciler/pkg/model"
 )
 
+type FilterMixer struct {
+	Filters []Filter
+}
+
+func (fm *FilterMixer) FilterByQuery(q *db.Select) error {
+	for i := range fm.Filters {
+		if err := fm.Filters[i].FilterByQuery(q); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (fm *FilterMixer) FilterByInstance(re *model.ReconciliationEntity) *model.ReconciliationEntity {
+	entity := re
+	for i := range fm.Filters {
+		if entity == nil {
+			break
+		}
+		entity = fm.Filters[i].FilterByInstance(entity)
+	}
+	return entity
+}
+
+type Limit struct {
+	Count       int
+	actualCount int
+}
+
+func (l *Limit) FilterByQuery(q *db.Select) error {
+	q.Limit(l.Count)
+	return nil
+}
+
+func (l *Limit) FilterByInstance(re *model.ReconciliationEntity) *model.ReconciliationEntity {
+	if l.actualCount < l.Count {
+		l.actualCount++
+		return re
+	}
+	return nil
+}
+
+type WithStatuses struct {
+	Statuses []string
+}
+
+func (ws *WithStatuses) FilterByQuery(q *db.Select) error {
+	if len(ws.Statuses) < 1 {
+		return nil
+	}
+
+	var whereRaw string
+	for i := range ws.Statuses {
+		if i > 0 {
+			whereRaw = fmt.Sprintf("%s%s", whereRaw, " OR ")
+		}
+
+		whereRaw = fmt.Sprintf("%sstatus='%s'", whereRaw, ws.Statuses[i])
+	}
+
+	q.WhereRaw((whereRaw))
+
+	return nil
+}
+
+func (ws *WithStatuses) FilterByInstance(re *model.ReconciliationEntity) *model.ReconciliationEntity {
+	if len(ws.Statuses) < 1 {
+		return nil
+	}
+
+	for i := range ws.Statuses {
+		if ws.Statuses[i] == string(re.Status) {
+			return re
+		}
+	}
+	return nil
+}
+
+type WithCreationDateAfter struct {
+	Time time.Time
+}
+
+func (wd *WithCreationDateAfter) FilterByQuery(q *db.Select) error {
+	q.WhereRaw(fmt.Sprintf("created>'%s'", wd.Time.Format("2006-01-02 15:04:05.000")))
```

Comment:
(as above) Please use db.ColumnHandler to retrieve column-names. The column handler ensures, that the column-name is always in-sync with the DB schema